src/common/config_proxy.h

```diff
@@ -344,6 +344,10 @@ class ConfigProxy {
   const std::string& get_conf_path() const {
     return config.get_conf_path();
   }
+  int get_val_default(const std::string_view key, std::string *val) {
+    std::lock_guard l{lock};
+    return config.get_val_default(key, val);
```

Comment:
A general question: must we lock when accessing defaults?
Can we create a faster, non lock, alternative for that?

Reply:
In this case, we don't need a lock since the value being read is the daemon default and is read-only. This is not a part of the "values" map maintained by the ConfigValues object that stores the alternate defaults and other config options that are modified during runtime.